rust/src/interpreter.rs

```diff
@@ -796,10 +722,16 @@ impl<'a> Interpreter<'a> {
                     }
 
                     self.step_status_code = StepStatusCode::EVMC_STEP_STOPPED;
-                    self.code_reader.next();
                     break;
                 }
             }
+
+            if !(Opcode::Push1 as u8..=Opcode::Push32 as u8).contains(&(op as u8))
```

Comment:
it is pretty safe to assume that push1-32 are sequential op code values. if you want to turn this into a comparison. 
Nevertheless I am curious, how good can rustc optimize this code ?

Reply:
I just checked. It's optimized away...